staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -169,6 +174,8 @@ func NewCustomResourceDefinitionHandler(
 	return ret, nil
 }
 
+var longRunningFilter = genericfilters.BasicLongRunningRequestCheck(sets.NewString("watch"), sets.NewString())
```

Comment:
Comment here on why tearing down rest storage in the presence of watches is safe?

Reply:
added a comment. for what it's worth, I have an expanded version of this test locally that also starts watches to make sure they get terminated cleanly. that recreates the watch hang bug. once this merges, I'll get that in a PR for use in verifying the watch hang when we fix that